src/main/java/databaseInteracting/DataBaseInteracting.java

```diff
@@ -0,0 +1,173 @@
+   * @param password String - password to login on the database.
+   * @throws SQLException - Exception of database
+   */
+  public DataBaseInteracting(String url, String user, String password) throws SQLException {
+    DataBaseConnection dataBaseConnection =
+        new DatabaseInteracting.DataBaseConnection(url, user, password);
+    connection = dataBaseConnection.getConnection();
+  }
+
+  /**
+   * Method to prepare the query.
+   *
+   * @return preparedStatement of the query to list all products.
+   * @throws SQLException - Exception of database.
+   */
+  public PreparedStatement createPrepareStatementToProductList() throws SQLException {
+    String sql = "SELECT * FROM products";
+    return connection.prepareStatement(sql);
+  }
+
+  /**
+   * Method to add a product on the products table.
+   *
+   * @param name  String - name of the product.
+   * @param stock int - number of products on the stock.
+   * @param price float - price of the product.
+   * @throws SQLException - Exception of database.
+   */
+  public void addProductToDatabase(String name, int stock, float price) throws SQLException {
+    String sql = "INSERT INTO products (name, stock, price) VALUES (?,?,?)";
+    PreparedStatement preparedStatement = connection.prepareStatement(sql);
+    preparedStatement.setString(1, name);
+    preparedStatement.setInt(2, stock);
+    preparedStatement.setFloat(3, price);
+    preparedStatement.executeUpdate();
+    preparedStatement.close();
+  }
+
+  /**
+   * Method to get products by the id.
+   *
+   * @param id int - id of the product.
+   * @return a resultSet with the products.
+   * @throws SQLException - Exception of database.
+   */
+  public ResultSet getProductsById(int id) throws SQLException {
+    PreparedStatement preparedStatement =
+        connection.prepareStatement("select * from products where id = ?");
+    preparedStatement.setInt(1, id);
+    return preparedStatement.executeQuery();
+  }
+
+  /**
+   * Method to get a resultSet of products table passing a Prepared Statement.
+   *
+   * @param preparedStatement PreparedStatement -
+   *                          prepared statement of the query to list all products.
+   * @return resultSet with data of the products table.
+   * @throws SQLException - Exception of database.
+   */
+  public ResultSet createResultSet(PreparedStatement preparedStatement) throws SQLException {
+    return preparedStatement.executeQuery();
+  }
+
+  /**
+   * Method to return all products into a Products table.
+   *
+   * @return list of products.
+   * @throws SQLException Exception of database.
+   */
+  public List<databaseInteracting.Product> searchForAllProductsOnDatabase() throws SQLException {
+    List<databaseInteracting.Product> list = new ArrayList<>();
+    try (PreparedStatement preparedStatement = createPrepareStatementToProductList();
+         ResultSet aResultSet = createResultSet(preparedStatement)) {
+      PrepareAndExecuteMethod(aResultSet, list);
```

Comment:
Methods in java, should not start with Capital Letter

Reply:
Done.